### PR TITLE
Spring Boot 2.2

### LINKED ISSUE #10565: Update spring-boot to 2.2 and spring-cloud to Hoxton
From spring calender: http://spring-calendar.cfapps.io/

- Spring-boot 2.2 is expected on [October 16th](https://github.com/spring-projects/spring-boot/milestones): https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.2-Release-Notes

- Spring-cloud Hoxton is expected on [November 7th](https://github.com/spring-cloud/spring-cloud-release/milestones) : https://github.com/spring-projects/spring-cloud/wiki/Spring-Cloud-Hoxton-Release-Notes

Created branch https://github.com/jhipster/generator-jhipster/tree/spring-boot_2.x and https://github.com/jhipster/jhipster/tree/spring-boot_2.x for migration.

For reference:
- https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-5.x#upgrading-to-version-52
- https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.2.0-Draft-Release-Notes

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by murdos**
Fixes #10565
I'm opening the PR to provide more visibility on the ongoing work, and on failing tests.

Currently we have errors with:
-   [x] Oauth2: there are issues related to mapping of GrantedAuthority
-   [x] Elasticsearch. Issue is related to ES using more core API of Log4j2 (see elastic/elasticsearch#19415). https://github.com/ctron/de.dentrassi.elasticsearch.log4j2-mock/pull/2 has been created to fix it.
-   [x] Mongo: use of both `@Field("event_id")` and `@Id` in `PersistentAuditEvent` is now problematic
-   [x] Kafka: error in `KafkaResourceIT.consumesMessages()`: `Content type not set`
-   [x] Issues with new version of Liquibase => reverted to 3.6.3

Please make sure the below checklist is followed for Pull Requests.
-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by murdos** [comment]
@mraible : could you help with the OAuth issue, please?
https://github.com/jhipster/generator-jhipster/blob/7d21a024c4c95f6f997e5155b155c3b2eb34a5df/generators/server/templates/src/main/java/package/config/SecurityConfiguration.java.ejs#L325-L331
`authority` is now an `SimpleGrantedAuthority` and not an `OidcUserAuthority` anymore, and the `ROLE_ADMIN` is missing (`authorities` only contain `ROLE_USER` and `SCOPE_*` for the admin user)

**Comment by DanielFran** [comment]
@mraible See also in jhipster lib: https://travis-ci.org/jhipster/jhipster/jobs/625029175#L346
````sh
[WARNING] /home/travis/build/jhipster/jhipster/jhipster-framework/src/main/java/io/github/jhipster/security/uaa/UaaAutoConfiguration.java:[28,69] org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails in org.springframework.security.oauth2.client.token.grant.client has been deprecated

[WARNING] /home/travis/build/jhipster/jhipster/jhipster-framework/src/main/java/io/github/jhipster/security/uaa/LoadBalancedResourceDetails.java:[26,69] org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails in org.springframework.security.oauth2.client.token.grant.client has been deprecated
````

**Comment by juliensadaoui** [comment]
Spring Cloud Hoxton SR1 is available.

https://spring.io/blog/2019/12/21/spring-cloud-hoxton-service-release-1-sr1-is-available

**Comment by mraible** [comment]
> @mraible See also in jhipster lib: travis-ci.org/jhipster/jhipster/jobs/625029175#L346

I don't think we can fix this one. 

> [WARNING] .../security/uaa/UaaAutoConfiguration.java:[28,69] 
> org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails 
> in org.springframework.security.oauth2.client.token.grant.client has been deprecated


This is using a feature that is no longer supported in Spring Security 5.2. From https://github.com/spring-projects/spring-security/wiki/OAuth-2.0-Migration-Guide:

> This document contains guidance for moving OAuth 2.0 Clients and Resource Servers from Spring Security OAuth 2.x to Spring Security 5.2.x. Since Spring Security doesn’t provide Authorization Server support, migrating a Spring Security OAuth Authorization Server is out of scope for this document.

In other words, Spring Security no longer supports building an Authorization Server, which is what UAA does. They might change their minds because of user feedback though. See [this blog post](https://spring.io/blog/2019/11/14/spring-security-oauth-2-0-roadmap-update#no-authorization-server-support) for more information.

@jgrandja Can you confirm that `ClientCredentialsResourceDetails` has been deprecated in Spring Security 5.2 and there's no workaround?

Related to JHipster's Spring Security 5.2 upgrade, how do we upgrade this code? See https://github.com/jhipster/generator-jhipster/pull/10955#issuecomment-565944069 for the runtime exception we're seeing.

```java
@Bean
public GrantedAuthoritiesMapper userAuthoritiesMapper() {
    return (authorities) -> {
        Set<GrantedAuthority> mappedAuthorities = new HashSet<>();

        authorities.forEach(authority -> {
            OidcUserAuthority oidcUserAuthority = (OidcUserAuthority) authority;
            mappedAuthorities.addAll(SecurityUtils.extractAuthorityFromClaims(oidcUserAuthority.getUserInfo().getClaims()));
        });
        return mappedAuthorities;
    };
}
```

**Comment by jgrandja** [comment]
@mraible 

> Can you confirm that `ClientCredentialsResourceDetails` has been deprecated in Spring Security 5.2 and there's no workaround?

`ClientCredentialsResourceDetails` lives in Spring Security OAuth 2.x NOT Spring Security 5.2. And yes it is deprecated just like all the classes in the project as part of the 2.4.0 release.

**Comment by mraible** [comment]
I'm trying to fix OAuth issues today and I noticed that even though I have Spring Boot 2.2.2.RELEASE in my pom.xml, it's pulling in 2.1.11.

```xml
<spring-boot.version>2.2.2.RELEASE</spring-boot.version>
```

```
mvn dependency:tree | grep spring-boot

...

INFO] +- org.springframework.boot:spring-boot-test:jar:2.1.11.RELEASE:test
[INFO] |  \- org.springframework.boot:spring-boot:jar:2.1.11.RELEASE:compile
```

**Comment by mraible** [comment]
To fix this, I had to clone the `jhipster` project and install it.

```
git clone git@github.com:jhipster/jhipster.git
cd jhipster
./mvnw clean install -Dgpg.skip=true
```

Then I had to update my `pom.xml` to point to it.

```xml
<jhipster-dependencies.version>3.2.0-SNAPSHOT</jhipster-dependencies.version>
```

**Comment by DanielFran** [comment]
@pascalgrimaud Should we merge it?

**Comment by pascalgrimaud** [comment]
We need to upgrade the jhipster lib and merge there first
Then, update this branch to use master of jhipster lib, then, if the CI pass, let's merge this for the next release !

**Comment by pascalgrimaud** [comment]
I think with Spring Boot 2.2, there is no breaking change, right ?
If there is no breaking change, it can be included in v6.7.0
Otherwise, I'll start to create v6.x and let's work on v7

**Comment by DanielFran** [comment]
No breaking change, just upgrade of dependencies...

**Comment by DanielFran** [comment]
@pascalgrimaud Should be awesome to have 6.7 in beginning of New Year

File: generators/server/templates/src/test/java/package/web/rest/AccountResourceIT.java.ejs [hunk 1]
```diff
@@ -204,7 +204,7 @@ public class AccountResourceIT <% if (databaseType === 'cassandra') { %>extends
         mockMvc.perform(get("/api/account")
             .accept(MediaType.APPLICATION_JSON))
             .andExpect(status().isOk())
-            .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON_VALUE))
```
Comment (murdos) [inline generators/server/templates/src/test/java/package/web/rest/AccountResourceIT.java.ejs]: I don't think indentation change is wanted here, probably a typo?

**Comment by DanielFran** [comment]
Thanks @murdos and @mraible

**Pull request merged**